Problem with not found OS images in ISO files provided

Gents,

---------------------------------------------------------------------------------------------------------------------------------
Problem with not found OS images in ISO files provided, as Get-LabAvailableOperatingSystem returns 0 images on my Microsoft Windows 10 Pro, 10.0.19042.0' (Polish).
The same code run on Windows 2016 runs smoothly (LabSources folder contains the same data).
I have created new account with English regional seettings, but it doesn't help.
---------------------------------------------------------------------------------------------------------------------------------

```
PS C:\WINDOWS\system32> C:\LabSources\Sample_Scripts_Mariusz\Scenarios\5_HyperVSCVMM_S2D_cluster_v1.0.ps1
12:10:55|00:00:00|00:00:00.000| Initialization
12:10:56|00:00:00|00:00:00.290| - Host operating system version: 'Microsoft Windows 10 Pro, 10.0.19042.0'
12:10:56|00:00:00|00:00:00.297| - Creating new lab definition with name 'MLAB1SCVMM'
12:10:56|00:00:00|00:00:00.340| - Location of lab definition files will be 'C:\ProgramData\AutomatedLab/Labs/MLAB1SCVMM'
12:10:56|00:00:00|00:00:00.905| - Location of LabSources folder is 'C:\LabSources'
12:10:57|00:00:01|00:00:00.001| - Auto-adding ISO files
12:10:58|00:00:02|00:00:00.251|   - found 1 OS images in the cache
12:10:58|00:00:02|00:00:00.259|   - Scanning 10 files for operating systemsFound 0 OS images.
12:11:03|00:00:07|00:00:05.765|   - Added 'C:\LabSources\ISOs\mu_system_center_configuration_manager_endpoint_protection_version_1606_x86_x64_dvd_9678361.iso'
12:11:03|00:00:07|00:00:05.781|   - Added 'C:\LabSources\ISOs\mu_system_center_operations_manager_2019_x64_dvd_b3488f5c.iso'
12:11:03|00:00:07|00:00:05.787|   - Added 'C:\LabSources\ISOs\mu_system_center_virtual_machine_manager_2019_x64_dvd_06c18108.iso'
12:11:03|00:00:07|00:00:05.791|   - Added 'C:\LabSources\ISOs\SQLServer2017-x64-ENU.iso'
12:11:03|00:00:07|00:00:05.799|   - Added 'C:\LabSources\ISOs\SW_DVD9_NTRL_SQL_Svr_Ent_Core_2017_64Bit_English_OEM_VL_X21-56995.ISO'
12:11:03|00:00:07|00:00:05.804|   - Added 'C:\LabSources\ISOs\SW_DVD9_SQL_Svr_Enterprise_Edtn_2019Dec2019_64Bit_English_MLF_X22-22247.ISO'
12:11:03|00:00:07|00:00:05.809|   - Added 'C:\LabSources\ISOs\SW_DVD9_Win_Pro_10_1909_64BIT_English_Pro_Ent_EDU_N_MLF_X22-17395.ISO'
12:11:03|00:00:07|00:00:05.815|   - Added 'C:\LabSources\ISOs\W2016_eval_Datacenter_EVAL_en-us_14393_refresh.ISO'
12:11:03|00:00:07|00:00:05.822|   - Added 'C:\LabSources\ISOs\W2019_en_windows_server_version_20h2_updated_dec_2020_x64_dvd_4b0c4bb4.iso'
12:11:03|00:00:07|00:00:05.827|   - Added 'C:\LabSources\ISOs\W2019_eval_17763.737.190906-2324.rs5_release_svc_refresh_SERVER_EVAL_x64FRE_en-us_1.iso'
12:11:03|00:00:07|00:00:05.844|   - Done
12:11:03|00:00:07|00:00:05.861| - Path for VMs specified as 'E:\HYPER-V\AutomatedLab-VMs\MLAB1SCVMM'
12:11:19|00:00:23|00:00:21.293| - Added 'C:\LabSources\ISOs\SW_DVD9_NTRL_SQL_Svr_Ent_Core_2017_64Bit_English_OEM_VL_X21-56995.ISO'
12:11:19|00:00:23|00:00:22.160| - Added 'C:\LabSources\ISOs\mu_system_center_virtual_machine_manager_2019_x64_dvd_06c18108.iso'
12:11:23|00:00:27|00:00:00.000| - Adding Hyper-V machine definition 'MLAB1DC01'   (Roles: RootDC)
The operating system 'Windows Server 2019 Datacenter Evaluation (Desktop Experience)' for machine 'MLAB1DC01' could not be found in the available operating systems. Call 'Get-LabAvailableOperatingSystem' to get a list of operating systems added to the lab.
At C:\Program Files\WindowsPowerShell\Modules\AutomatedLabDefinition\5.32.0\AutomatedLabDefinition.psm1:2835 char:21
+ ...             throw "The operating system '$OperatingSystem' for machin ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (The operating s...ded to the lab.:String) [], RuntimeException
    + FullyQualifiedErrorId : The operating system 'Windows Server 2019 Datacenter Evaluation (Desktop Experience)' for machine 'MLAB1DC01' could not be found in the available operating systems. Call 'Get-LabAvailableOperatingSystem' to get a list of operating systems ad 
   ded to the lab.
```

---------------------------------------------------------------------------------------------------------------------------------
Results of Get-LabAvailableOperatingSystem:
---------------------------------------------------------------------------------------------------------------------------------

```
PS C:\WINDOWS\system32> Get-LabAvailableOperatingSystem -Verbose
VERBOSE: [12:18:58][Write-LogFunctionEntry] Get-LabAvailableOperatingSystem Entering... (Verbose=True)
VERBOSE: [12:18:58][Get-LabAvailableOperatingSystem] Found 10 ISO files
VERBOSE: [12:18:58][Write-ScreenInfo] found 1 OS images in the cache
VERBOSE: [12:18:58][Write-ScreenInfo] 12:18:58|00:08:02|00:07:35.434| found 1 OS images in the cache
12:18:58|00:08:02|00:07:35.434| found 1 OS images in the cache
VERBOSE: [12:18:58][Write-ScreenInfo] Scanning 10 files for operating systems
12:18:58|00:08:03|00:07:35.443| Scanning 10 files for operating systemsVERBOSE: [12:18:58][Write-ScreenInfo] 12:18:58|00:08:03|00:07:35.443| Scanning 10 files for operating systems
VERBOSE: [12:18:58][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\mu_system_center_configuration_manager_endpoint_protection_version_1606_x86_x64_dvd_9678361.iso'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\mu_system_center_operations_manager_2019_x64_dvd_b3488f5c.iso'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\mu_system_center_virtual_machine_manager_2019_x64_dvd_06c18108.iso'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\SQLServer2017-x64-ENU.iso'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\SW_DVD9_NTRL_SQL_Svr_Ent_Core_2017_64Bit_English_OEM_VL_X21-56995.ISO'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\SW_DVD9_SQL_Svr_Enterprise_Edtn_2019Dec2019_64Bit_English_MLF_X22-22247.ISO'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\SW_DVD9_Win_Pro_10_1909_64BIT_English_Pro_Ent_EDU_N_MLF_X22-17395.ISO'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\W2016_eval_Datacenter_EVAL_en-us_14393_refresh.ISO'
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\W2019_en_windows_server_version_20h2_updated_dec_2020_x64_dvd_4b0c4bb4.iso'
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\W2019_eval_17763.737.190906-2324.rs5_release_svc_refresh_SERVER_EVAL_x64FRE_en-us_1.iso'
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:19:00][Write-ScreenInfo] Found 0 OS images.
Found 0 OS images.
VERBOSE: [12:19:00][Write-LogFunctionExit] Get-LabAvailableOperatingSystem...leaving...(Time elapsed: 00:00:02:490)PS C:\WINDOWS\system32> Get-LabAvailableOperatingSystem -Verbose
VERBOSE: [12:18:58][Write-LogFunctionEntry] Get-LabAvailableOperatingSystem Entering... (Verbose=True)
VERBOSE: [12:18:58][Get-LabAvailableOperatingSystem] Found 10 ISO files
VERBOSE: [12:18:58][Write-ScreenInfo] found 1 OS images in the cache
VERBOSE: [12:18:58][Write-ScreenInfo] 12:18:58|00:08:02|00:07:35.434| found 1 OS images in the cache
12:18:58|00:08:02|00:07:35.434| found 1 OS images in the cache
VERBOSE: [12:18:58][Write-ScreenInfo] Scanning 10 files for operating systems
12:18:58|00:08:03|00:07:35.443| Scanning 10 files for operating systemsVERBOSE: [12:18:58][Write-ScreenInfo] 12:18:58|00:08:03|00:07:35.443| Scanning 10 files for operating systems
VERBOSE: [12:18:58][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\mu_system_center_configuration_manager_endpoint_protection_version_1606_x86_x64_dvd_9678361.iso'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\mu_system_center_operations_manager_2019_x64_dvd_b3488f5c.iso'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\mu_system_center_virtual_machine_manager_2019_x64_dvd_06c18108.iso'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\SQLServer2017-x64-ENU.iso'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\SW_DVD9_NTRL_SQL_Svr_Ent_Core_2017_64Bit_English_OEM_VL_X21-56995.ISO'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\SW_DVD9_SQL_Svr_Enterprise_Edtn_2019Dec2019_64Bit_English_MLF_X22-22247.ISO'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\SW_DVD9_Win_Pro_10_1909_64BIT_English_Pro_Ent_EDU_N_MLF_X22-17395.ISO'
VERBOSE: [12:18:59][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\W2016_eval_Datacenter_EVAL_en-us_14393_refresh.ISO'
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\W2019_en_windows_server_version_20h2_updated_dec_2020_x64_dvd_4b0c4bb4.iso'
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Mounting ISO image 'C:\LabSources\ISOs\W2019_eval_17763.737.190906-2324.rs5_release_svc_refresh_SERVER_EVAL_x64FRE_en-us_1.iso'
VERBOSE: [12:19:00][Get-LabAvailableOperatingSystem] Dismounting ISO
VERBOSE: [12:19:00][Write-ScreenInfo] Found 0 OS images.
Found 0 OS images.
VERBOSE: [12:19:00][Write-LogFunctionExit] Get-LabAvailableOperatingSystem...leaving...(Time elapsed: 00:00:02:490)
```

---------------------------------------------------------------------------------------------------------------------------------

PS C:\WINDOWS\system32> $PSVersionTable
```
Name                           Value                                                                                                                                                                                                                                            
----                           -----                                                                                                                                                                                                                                            
PSVersion                      5.1.19041.610                                                                                                                                                                                                                                    
PSEdition                      Desktop                                                                                                                                                                                                                                          
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                                          
BuildVersion                   10.0.19041.610                                                                                                                                                                                                                                   
CLRVersion                     4.0.30319.42000                                                                                                                                                                                                                                  
WSManStackVersion              3.0                                                                                                                                                                                                                                              
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                                              
SerializationVersion           1.1.0.
```
---------------------------------------------------------------------------------------------------------------------------------

PS C:\WINDOWS\system32> Get-Module AutomatedLab* -List | Select Name,Version
```
Name                      Version
----                      -------
AutomatedLab              5.32.0 
AutomatedLab.Common       2.0.208
AutomatedLabDefinition    5.32.0 
AutomatedLabNotifications 5.32.0 
AutomatedLabUnattended    5.32.0 
AutomatedLabWorker        5.32.0
```

---------------------------------------------------------------------------------------------------------------------------------

After mounting ISO

Dism.exe /English /Get-WimInfo /WimFile:D:\sources\install.wim

```
PS C:\WINDOWS\system32> Dism.exe /English /Get-WimInfo /WimFile:D:\sources\install.wim

Deployment Image Servicing and Management tool
Version: 10.0.19041.746

Details for image : D:\sources\install.wim

Index : 1
Name : Windows Server 2019 Standard Evaluation
Description : (Recommended) This option omits most of the Windows graphical environment. Manage with a command prompt and PowerShell, or remotely with Windows Admin Center or other tools.
Size : 8˙230˙200˙012 bytes

Index : 2
Name : Windows Server 2019 Standard Evaluation (Desktop Experience)
Description : This option installs the full Windows graphical environment, consuming extra drive space. It can be useful if you want to use the Windows desktop or have an app that requires it.
Size : 14˙445˙738˙585 bytes

Index : 3
Name : Windows Server 2019 Datacenter Evaluation
Description : (Recommended) This option omits most of the Windows graphical environment. Manage with a command prompt and PowerShell, or remotely with Windows Admin Center or other tools.
Size : 8˙233˙207˙447 bytes

Index : 4
Name : Windows Server 2019 Datacenter Evaluation (Desktop Experience)
Description : This option installs the full Windows graphical environment, consuming extra drive space. It can be useful if you want to use the Windows desktop or have an app that requires it.
Size : 14˙447˙276˙648 bytes

The operation completed successfully.
```

@raandree , using quotes this time, thanks for making up my previous mess ...

So far I have detected that the line from AutomatedLabDisks.psm1 fails:

` $diskImage | Add-Member -MemberType NoteProperty -Name DriveLetter -Value ($diskImage | Get-Volume).DriveLetter -PassThru
`

The disk image is attached now and in Windows Explorer I can see its drive letter as well as when I use Get-PSdrive - in this case it's F:

```
PS E:\HYPER-V\Wslab\ISOFiles> $diskimage

Attached          : True
BlockSize         : 0
DevicePath        : \\.\CDROM1
FileSize          : 5296713728
ImagePath         : E:\HYPER-V\Wslab\ISOFiles\17763.737.190906-2324.rs5_release_svc_refresh_W2019_SERVER_EVAL_x64FRE_en-us_1.iso
LogicalSectorSize : 2048
Number            : 1
Size              : 5296713728
StorageType       : 1
PSComputerName    : 


PS E:\HYPER-V\Wslab\ISOFiles> Get-Volume -DiskImage $diskimage

PS E:\HYPER-V\Wslab\ISOFiles> Get-PSDrive

Name           Used (GB)     Free (GB) Provider      Root                                                                                                                                                CurrentLocation
----           ---------     --------- --------      ----                                                                                                                                                ---------------
Alias                                  Alias                                                                                                                                                                            
C                 195,93         41,07 FileSystem    C:\                                                                                                                                                WINDOWS\system32
Cert                                   Certificate   \                                                                                                                                                                  
D                 257,31        674,06 FileSystem    D:\                                                                                                                                                                
E                 382,34        571,51 FileSystem    E:\                                                                                                                                          HYPER-V\Wslab\ISOFiles
Env                                    Environment                                                                                                                                                                      
F                   4,93          0,00 FileSystem    F:\ 
```

But as you can see

`PS E:\HYPER-V\Wslab\ISOFiles> Get-Volume -DiskImage $diskimage`

returns empty string as well as:

`PS E:\HYPER-V\Wslab\ISOFiles> ($diskimage | Get-Volume).DriveLetter`

I need to go deeper into my Windows :-)

--- a/AutomatedLab/AutomatedLab.psm1
+++ b/AutomatedLab/AutomatedLab.psm1
@@ -30,9 +30,18 @@ function Enable-LabHostRemoting
     {
         Write-ScreenInfo 'Starting the WinRM service. This is required in order to read the WinRM configuration...' -NoNewLine
         Start-Service -Name WinRM
-        Start-Sleep -Seconds 5
+        Start-Sleep -Seconds 2
         Write-ScreenInfo done
     }
+
+    if ((Get-Service -Name smphost).StartType -eq 'Disabled')
+    {
+        Write-ScreenInfo "The StartupType of the service 'smphost' is set to disabled. Setting it to 'manual'. This is required in order to read use the cmdlets in the 'Storage' module..." -NoNewLine
+        Set-Service -Name smphost -StartupType Manual
+        Write-ScreenInfo done
+    }
+
+    #1067
 
     # force English language output for Get-WSManCredSSP call
     [Threading.Thread]::CurrentThread.CurrentUICulture = 'en-US'; $WSManCredSSP = Get-WSManCredSSP

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@
 
 - Fixed issue with CentOS8/RHEL8 kickstart files being undeployable (issue #1065)
 - Fixed issue with International module on PS Core not being imported (issue #1066)
+- Fixed issue when cmdlets of the 'Storage' module did not work if the 'smphost' is set to disabled (issue #1067)
 
 ## 5.32.0 (2020-12-31)
 

